Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform Brick;
+    private int[,] Maze;
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> _tiletoTry = new Stack<Vector2>();
+    private List<Vector2> offsets = new List<Vector2> { new Vector2(0, 1), new Vector2(0, -1), new Vector2(1, 0), new Vector2(-1, 0) };
+    private System.Random rnd = new System.Random();
+    private int _width, _height;
+    private Vector2 _currentTile;
+    public Vector2 CurrentTile
+    {
+        get
+        {
+            return _currentTile;
+        }
+
+        private set
+        {
+            if (value.x < 1 || value.x >= this.width - 1 || value.y < 1 || value.y >= this.height - 1)
+            {
+                throw new ArgumentException("CurrentTile must be within the one tile border all around the maze");
+            }
+            if (value.x % 2 == 1 || value.y % 2 == 1)
+            {
+                _currentTile = value;
+            }
+            else
+            {
+                throw new ArgumentException("The current square must not be both on an even X-axis and an even Y-axis, to ensure we can get walls around all tunnels");
+            }
+        }
+    }
+    public static MazeGenerator Instance { get; private set; }
+
+    private void Awake()
+    {
+        Instance = this;
+        GenerateMaze();
+    }
+
+    private void Start()
+    {
+
+    }
+
+    private void GenerateMaze()
+    {
+        Maze = new int[width, height];
+        for (int x = 0; x < width; x++)
+        {
+            for (int y = 0; y < height; y++)
+            {
+                Maze[x, y] = 1;
```

Comment:
What is this magic number 1? Consider using an `Enum` instead to be more descriptive and improve code readability.